### PR TITLE
Support for job-type `ros-provider-with-pydtk`

### PR DESCRIPTION
**Pull request by d-hayashi**
## What?
Support for job-type `ros-provider-with-pydtk`

## Why?
To use app-scene-viewer

## See also [Optional]
https://github.com/dataware-tools/api-job-store/pull/26
https://github.com/dataware-tools/protocols/pull/52

## Screenshot or video [Optional]

https://user-images.githubusercontent.com/24401842/145756976-be0fce9c-be0c-4e56-a170-9fcf7f71be69.mov

### CODE REVIEW

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-data-browser-next/pull/250#" title="Author self-approved">d-hayashi</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-data-browser-next).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-data-browser-next/blob/master/OWNERS)~~ [d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/components/molecules/FilePreviewer/index.tsx
Comment (WatanabeToshimitsu) [inline src/components/molecules/FilePreviewer/index.tsx:21]: Javascript/Typescript って関数の引数の扱いが，python に比べて少し面倒だと思うんですが，引数としてオブジェクトを受け取ることで扱いを楽にできたりします

今くらい単純な関数だと特に必要性が感じられないですが，`FilePreviewerContentWithSpec` 型の定義を
```
～～～
render: (`{databaseId: string, recordId: string, file: FileType}) => JSX.Element;
```
とすると，
`filePreviewerCandates` の各要素の `render` プロパティを
```
render: ({databaseId, file}) => hoge
```
と記述できるようになります

普通に関数を定義して使うにしても，より明示的に引数を渡せるようになったり，オプショナルな引数に対して undefined をわざわざ渡す必要もなくなるので，結構便利かなと思ってます

**Review by WatanabeToshimitsu — COMMENTED** [review]
LGTM です！
参考までに，ちょっとした Tips を書いときました

File: src/components/molecules/FilePreviewer/index.tsx
Comment (d-hayashi) [reply src/components/molecules/FilePreviewer/index.tsx]: なるほど！ Tipsあざます  

`render: (databaseId: string, recordId: string, file: FileType) => JSX.Element;`  
の部分を
`render: ({databaseId: string, recordId: string, file: FileType}) => JSX.Element;`  
という風に書き換えると↓の画像にあるTS2300で怒られてしまったので、下記のように変更しました    
スクリーンショット 2021-12-14 15 52 37


```typescript
type FilePreviewerContentRenderSpec = {
  databaseId: string,
  recordId: string,
  file: FileType,
}

type FilePreviewerContentWithSpec = {
  spec: {
    extensions: string[];
    contentTypes: string[];
  };
  render: ({databaseId, recordId, file}: FilePreviewerContentRenderSpec) => JSX.Element;
};
```

Comment (d-hayashi) [reply src/components/molecules/FilePreviewer/index.tsx]: この実装で問題無ければマージお願いします！

**Review by WatanabeToshimitsu — APPROVED** [review]
LGTM!

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-data-browser-next/pull/250#" title="Author self-approved">d-hayashi</a>*, *<a href="https://github.com/dataware-tools/app-data-browser-next/pull/250#pullrequestreview-832258425" title="Approved">WatanabeToshimitsu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-data-browser-next).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-data-browser-next/blob/master/OWNERS)~~ [WatanabeToshimitsu,d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
